scvi/module/_vae.py

```diff
@@ -360,8 +353,20 @@ def generative(
 
         px_r = torch.exp(px_r)
 
+        if self.gene_likelihood == "zinb":
+            px_latents = ZeroInflatedNegativeBinomial(
+                mu=px_rate, theta=px_r, zi_logits=px_dropout
+            )
+        elif self.gene_likelihood == "nb":
+            px_latents = NegativeBinomial(mu=px_rate, theta=px_r)
+        elif self.gene_likelihood == "poisson":
+            px_latents = Poisson(px_rate)
         return dict(
-            px_scale=px_scale, px_r=px_r, px_rate=px_rate, px_dropout=px_dropout
+            px_latents=px_latents,
```

Comment:
I think `px` is better here to be consistent that it's a distribution and not a parameter thereof

Reply:
i agree, i am on it